rfc/0011-audit-checks.md

```diff
@@ -113,6 +113,370 @@ resources of a given Namespace.
+
+#### Step 2: fetch Kubernetes resources to be audited
+
+After determining the policies for each resource type, the next task is to
+gather the instances of these resources present in the cluster. This is done by
+querying the API server, often using pagination to manage the data efficiently.
+
+Post-retrieval, the audit scanner assesses whether each resource aligns with
+the criteria set out in the policies, specifically focusing on the labels
+required by the policy. If a resource fits the criteria, the scanner simulates
+a `CREATE` event, incorporating the audited Kubernetes object as `request.object`
+within this event.
+
+### Step 3: performing evaluations
+
+Initially, the evaluation is executed through an HTTP request directed at the
+Policy Server that enforces the policy. This step is crucial for determining
+the compliance of each resource with the established cluster policies.
+
+We will extend Policy Server to have a new endpoint for the audit checks. While
+the Kubernetes API server uses `/validate/POL_ID`, the audit scanner will file
+its requests against `/audit/POL_ID` endpoint.
+
+The `/audit` endpoint will be different from the `/validate` one in these
+regards:
+
+* Generate different Prometheus metrics: we don't want the overall `/validate`
+  metrics to be polluted by the audit checks
+* Generate different Open Telemetry traces: again, these should not be mixed
+  with the `/validate` ones. We might even want to turn them off to make the
+  audit execution faster
+* Policies in `monitor` mode should still behave like the ones in `protect`
+  mode. They have to produce rejections if needed
+
+By issuing the audit request against a real Policy Server, we solve these
+problems:
+
+* Policies can be hosted on registries secured with credentials. We don't have
+  to worry about being able to fetch the WebAssembly modules from these
+  authenticated registries
+* Policies might be hosted on insecure registires/registires with custom TLS
+  certificates. Again, we won't have to bother about that
+* Some data used by the policy (like external OCI manifests, sigstore
+  verification results) can be cached by the Policy Server and reused between
+  the audit evaluations and the regular validations requests
+
+Sending all our auditing requests to the actual Policy Server could lead to
+poor performance of the regular admission requests. Currently, based on our
+load testing data, one single Policy Server running with 2 workers can evaluate
+about 1600 requests per second. Based on this data, we think the requests
+generated by the audit checks should not clog the Policy Server. If that turns
+out to be true, we can embrace different strategies to mitigate that.
+
+Until we have real world data, it's not useful to eagerly optimize this aspect.
+
+### Step 4: write `ClusterPolicyReport`
+
+The code will overwrite the old `ClusterPolicyReport` instance with a new generated
```

Comment:
We don't overwrite but we update the old object field with new data, also I would remove "data kept in memory" as we are keeping the data just for enough time to build the report patch.